src/Behat/Mink/WebAssert.php

```diff
@@ -590,11 +594,14 @@ public function fieldValueNotEquals($field, $value, TraversableElement $containe
     {
         $node   = $this->fieldExists($field, $container);
         $actual = $node->getValue();
-        $regex  = '/^'.preg_quote($value, '/').'$/ui';
 
-        $message = sprintf('The field "%s" value is "%s", but it should not be.', $field, $actual);
+        $message = sprintf(
+            'The field "%s" value is "%s", but it should not be.',
+            $field,
+            (is_array($actual) ? '['.implode(', ', $actual).']' : $actual)
+        );
 
-        $this->assert(!preg_match($regex, $actual), $message);
+        $this->assert($actual !== $value, $message);
```

Comment:
> `preg_quote(false, '/')` returns `''`, `preg_quote(true, '/')` returns `'1'`.

When boolean value is returned by `$node->getValue()`? If I'm not mistaken, then for checkboxes the returned value is `null` when non-checked and value of `value` attribute, when checked.

> These are the only two cases I found which might not work any longer.

Tests would indicate what no longer works after a change.

> I do not see, where else backwards compatibility could be broken.

1. the case sensitive comparison is done now (in PR) instead of case insensitive
2. the `===` is used now but maybe some drivers would return `4` as integer and others `4` as string and check will fail

> IMHO, there is no difference between the outcome of `'foo' === 'foo'` and `preg_match('^foo$', 'foo').`

Indeed, but there was is no `$` present currently and therefore it's not the same. The equivalent would be using `mb_substr`, but that would add dependency to `mbstring` PHP extension, which by default isn't present.

> `preg_quote(['foo', 'bar'], '/')` does not work and throws an error. This is fixed now.

When does `$node->getValue()` return an array? Probably for multi-select element, but I'm not sure that changed method is supposed to work with that element type.